src/plugin.js

```diff
@@ -223,9 +190,45 @@ const perSourceBehaviors = function() {
     VERSION: '__VERSION__'
   };
 
-  // Add the per-source event binding methods to this player.
-  this.onPerSrc = onPerSrc;
-  this.onePerSrc = onePerSrc;
+  /**
+   * Bind an event listener on a per-source basis.
+   *
+   * @function onPerSrc
+   * @param  {String|Array|Component|Element} first
+   *         The event type(s) or target Component or Element.
+   *
+   * @param  {Function|String|Array} second
+   *         The event listener or event type(s) (when `first` is target).
+   *
+   * @param  {Function} third
+   *         The event listener (when `second` is event type(s)).
+   *
+   * @return {Player}
+   */
+  this.onPerSrc = createPerSrcBinder();
+
+  /**
+   * Bind an event listener on a per-source basis. This listener can only
+   * be called once.
+   *
+   * @function onePerSrc
+   * @param  {String|Array|Component|Element} first
+   *         The event type(s) or target Component or Element.
+   *
+   * @param  {Function|String|Array} second
+   *         The event listener or event type(s) (when `first` is target).
+   *
+   * @param  {Function} third
+   *         The event listener (when `second` is event type(s)).
+   *
+   * @return {Player}
+   */
+  this.onePerSrc = createPerSrcBinder(true);
+
+  // Clear out perSrcListeners cache on player dispose.
+  this.on('dispose', () => {
+    perSrcListeners.length = 0;
```

Comment:
This actually won't work because there is a potential for other listeners to come before this one. It would work in our player, but we need to make sure this happens _before_ any `sourcechanged` listeners have a chance to add new per-source listeners.